### PR TITLE
feat(sidebar): unified activity rail + dockable panel system

### PR DESCRIPTION
**Pull request by sam-phinizy**
## Summary
- Replaces the ad-hoc right-side overlay panels with one **activity rail** (configurable to left/right edge) and a single **dockable panel region** that supports pin/active stacking. Sessions, Notes, Watches, Tasks, Docs, Notifications and Settings are all reachable from the rail.
- **Sessions** is folded into the dock as a first-class pinnable panel, pinned by default on first launch so the list behaves like a tab strip. Archived sessions collapse inside \`SessionTabs\` via a new \`ArchivedSessionsList\` component (the standalone \`SessionsPane\` is removed).
- Pin/split rendering: pinned panel on top, active below, draggable horizontal divider between them. Docs/Settings take over the full region when opened.

## UX model
- **Rail**: click = toggle; right-click or shift-click = pin/unpin; unread count badge on Notifications.
- **Pin button** visible in every pinnable panel's header (new \`PinButton.svelte\`) — no longer hidden behind right-click.
- **Unpinning the anchor wins**: if Sessions is pinned and Watches is active, unpinning Sessions promotes Sessions to full dock and drops Watches (not the other way around). Takeover panels (Docs/Settings) are preserved through unpin.
- \`Cmd+\\\` now hides the **entire** rail+dock (focus/distraction-free mode), not just the Sessions list.

## State + persistence (\`src/lib/stores/ui.ts\` + \`sidebarLayout.ts\`)
- Pin-slot hybrid: \`{ pinned, active }\`. Split rendering when both differ.
- localStorage persistence. Malformed JSON / \`NaN\` / \`Infinity\` / stringified numbers all fall back to defaults.
- Legacy \`dockSide\` migrates to \`railSide\`; removed \`sessionsShown\`, \`railPosition\`, \`tabPosition\` (UI side). Rust schema fields (\`sidebarCollapsed\`, \`taskPanelCollapsed\`, \`taskPanelSplit\`, \`tabPosition\`) are left intact so existing settings files don't break.

## Hardening from a deep Codex review
- Drag handlers cleanup on \`onDestroy\` / \`Escape\` / \`window.blur\` (covers lost-mouseup + mid-drag unmount).
- Split drag recomputes \`getBoundingClientRect()\` every \`mousemove\` instead of snapshotting at drag start (fixes stale-geometry bug when dock height changes during drag).
- \`clampWidth\` / \`clampRatio\` reject non-finite inputs defensively.

## Deleted
- \`CollapsedSidebar.svelte\` (no more collapse concept — Sessions is just a panel)
- \`SessionsPane.svelte\` (archive merged into \`SessionTabs\`)
- \`sidebarCollapsed\` / \`tabPosition\` / \`railPosition\` / \`dockSide\` settings UI
- Inline \`TaskPanel\` mount inside \`SessionTabs\`

## Tests
426 passing, 0 errors, 0 warnings. Added 15 new tests:
- Persistence corruption (malformed JSON, NaN, Infinity, stringified numbers, out-of-range clamping)
- Unpin edge cases (docs takeover, settings takeover, anchor promotion)
- Rail interactions (right-click pin/unpin, non-pinnable icons, takeover sequences)

## Test plan
- [ ] \`task dev\` — rail renders on the default (right) side, Sessions list visible
- [ ] Click each rail icon — panel opens in the dock, clicking again closes
- [ ] Pin Notes, click Watches — both visible stacked with draggable divider
- [ ] Pin Notes, click Docs — Docs fills the dock, close Docs → Notes returns pinned
- [ ] Drag dock right/left edge — width persists across restart
- [ ] Settings → General → switch \"Sidebar position\" to Left — whole chrome flips
- [ ] \`Cmd+\\\` — rail + dock both hide; press again → both return with prior state
- [ ] Open the Archived section at the bottom of SessionTabs — Restore / Notes / Reveal / Clean / Delete all work inline
- [ ] Kill the app and relaunch — pinned panel + rail side + dock width + hidden flag all restored

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by sam-phinizy** [comment]
@codex review please

File: src/lib/components/SidebarDock.svelte [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  function onCloseFor(id: SidebarId): () => void {
+    // If the panel is pinned AND in the active slot, closing should just clear
+    // the active slot (not unpin). If it's only pinned, close should unpin.
+    return () => {
+      if (pinned === id && active !== id) {
+        unpinSidebar();
+      } else {
+        closeSidebar();
+      }
+    };
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/components/SidebarDock.svelte:183]: Good catch — this was a real bug. `unpinSidebar()` does anchor-promotion on purpose (the PinButton relies on it), but the × close button should just dismiss the panel, not promote.

Fixed in 8320910 by adding a separate `closePinned()` helper and routing the close button through it. `onCloseFor` now clears only whichever slot(s) hold the clicked panel. Added tests for both behaviors.

File: src/lib/commands/ui.ts [hunk 2]
```diff
@@ -211,34 +223,111 @@ export function registerUiCommands() {
...
+      if (pinned) {
+        unpinSidebar();
+        return;
+      }
+      const active = get(activeSidebar);
+      if (active && PINNABLE_SIDEBARS.has(active)) pinSidebar(active);
     },
   });
 
+  registry.register({
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/commands/ui.ts:316]: Correct — the Help menu item was silently broken when Cmd+\\ had hidden the rail+dock.

Fixed in 8320910: `help.open-docs` now calls `showSidebar()` before `openSidebar("docs")` so it works regardless of sidebar visibility.

File: src/lib/commands/ui.ts [hunk 3]
```diff
@@ -211,34 +223,111 @@ export function registerUiCommands() {
...
+    available: () => !!queries.activeSessionId(),
+    execute: () => toggleSidebar("tasks"),
+  });
+
+  registry.register({
+    id: "ui.toggle-tasks",
+    label: "Toggle Tasks Panel",
     category: "App",
     available: () => !!queries.activeSessionId(),
+    execute: () => toggleSidebar("tasks"),
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/commands/ui.ts]: Right, duplicate commands in the palette is bad. Removed `ui.toggle-tasks` in 8320910. `ui.toggle-task-panel` stays as the canonical id — it's already bound in the default keymap and `appMenu.ts`, so keeping it preserves existing bindings.

File: src/lib/components/SidebarDock.svelte [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    return `position: absolute; left: 0; right: 0; top: calc(${ratio * 100}% + 2px); bottom: 0;`;
+  }
+
+  function onCloseFor(id: SidebarId): () => void {
+    // If the panel is pinned AND in the active slot, closing should just clear
+    // the active slot (not unpin). If it's only pinned, close should unpin.
+    return () => {
+      if (pinned === id && active !== id) {
+        unpinSidebar();
+      } else {
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/components/SidebarDock.svelte]: Correct. Same root cause as #3127573715 — `unpinSidebar()` anchor-promotes (intentional for the pin button), but the close button shouldn't. Fixed in 8320910: new `closePinned()` helper that clears only the pinned slot, wired into `onCloseFor`. Tests added for solo-pinned and pinned+active close scenarios.
